pymc_experimental/inference/jax_find_map.py

```diff
@@ -0,0 +1,450 @@
+    X = pt.tensor("transformed_draws", shape=(chains, draws, H_inv.shape[0]))
+    out = []
+    for rv, idx in slices.items():
+        f = model.rvs_to_transforms[rv]
+        untransformed_X = f.backward(X[..., idx]) if f is not None else X[..., idx]
+
+        if rv in out_shapes:
+            new_shape = (chains, draws) + out_shapes[rv]
+            untransformed_X = untransformed_X.reshape(new_shape)
+
+        out.append(untransformed_X)
+
+    f_untransform = pytensor.function([X], out, mode="JAX")
+    return f_untransform(posterior_draws)
+
+
+def fit_laplace(
+    optimized_point: dict[str, np.ndarray],
+    model: pm.Model,
+    chains: int = 2,
+    draws: int = 500,
+    on_bad_cov: Literal["warn", "error", "ignore"] = "ignore",
+    transform_samples: bool = True,
+    zero_tol: float = 1e-8,
+    diag_jitter: float | None = 1e-8,
+    progressbar: bool = True,
+    mode: str = "JAX",
+) -> az.InferenceData:
+    """
+    Compute the Laplace approximation of the posterior distribution.
+
+    The posterior distribution will be approximated as a Gaussian
+    distribution centered at the posterior mode.
+    The covariance is the inverse of the negative Hessian matrix of
+    the log-posterior evaluated at the mode.
+
+    Parameters
+    ----------
+    optimized_point : dict[str, np.ndarray]
+        Local maximum a posteriori (MAP) point returned from pymc.find_MAP
+        or jax_tools.fit_map
+    model : Model
+        A PyMC model
+    chains : int
+        The number of sampling chains running in parallel. Default is 2.
+    draws : int
+        The number of samples to draw from the approximated posterior. Default is 500.
+    on_bad_cov : str, one of 'ignore', 'warn', or 'error', default: 'ignore'
+        What to do when ``H_inv`` (inverse Hessian) is not positive semi-definite.
+        If 'ignore' or 'warn', the closest positive-semi-definite matrix to ``H_inv`` (in L1 norm) will be returned.
+        If 'error', an error will be raised.
+    transform_samples : bool
+        Whether to transform the samples back to the original parameter space. Default is True.
+    zero_tol: float
+        Value below which an element of the Hessian matrix is counted as 0.
+        This is used to stabilize the computation of the inverse Hessian matrix. Default is 1e-8.
+    diag_jitter: float | None
+        A small value added to the diagonal of the inverse Hessian matrix to ensure it is positive semi-definite.
+        If None, no jitter is added. Default is 1e-8.
+    progressbar : bool
+        Whether or not to display progress bar. Default is True.
+    mode : str
+        Computation backend mode. Default is "JAX".
+
+    Returns
+    -------
+    InferenceData
+        arviz.InferenceData object storing posterior, observed_data, and constant_data groups.
+
+    """
+    frozen_model = freeze_dims_and_data(model)
+    if not transform_samples:
+        untransformed_model = remove_value_transforms(frozen_model)
```

Comment:
Are you share you want to remove the value transforms? That will take the optimizer out of an unconstrained space